layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -979,3 +979,17 @@ a split-side entry, its value must be usable as the SIDE argument for
                                             text-scale-mode-amount) 1)
                                   (if (car (window-margins))
                                       (car (window-margins)) 1)))))
+
+(defun spacemacs/linum-enabled-p ()
+  "Return non-nil if line numbers should be enabled for current buffer.
+Decision is based on `dotspacemacs-line-numbers'."
+  (if dotspacemacs-line-numbers
+      (if (or (string-match-p "\\*.*\\*" (buffer-name))
+              (and (plist-get dotspacemacs-line-numbers :size-limit-kb)
+		   (> (buffer-size) (* 1000 (plist-get dotspacemacs-line-numbers :size-limit-kb)))))
+	  nil
+        (if (plist-get dotspacemacs-line-numbers :enabled-for)
+            (member major-mode (plist-get dotspacemacs-line-numbers :enabled-for))
+          (and (plist-get dotspacemacs-line-numbers :disabled-for)
+               (not (member major-mode (plist-get dotspacemacs-line-numbers :disabled-for))))))
+    nil))
```

Comment:
I guess which one is harder to read depends on how your brain is wired. Or maybe I'm too much of an imperative person, but it was easier for me to read and build it that way.

I still will try out your suggestion though. Even if I find it harder in your example perhaps will it be otherwise with the real stuff.